### ISSUE TITLE
Bind to libyul instead of libsolc

### ISSUE THREAD

**Issue by g-r-a-n-t** [ethereum/fe#28]
We are currently binding to libsolc using [solc-rust](https://github.com/axic/solc-rust) and performing Yul -> bytecode compilation through the json interface.

This is problematic because building the whole Solidity compiler is very resource and time intensive (can take up to an hour). It would be nice if we could instead bind to [libyul](https://github.com/ethereum/solidity/tree/develop/libyul), as this takes a fraction of the time to build.

**Comment by cburgdorf** [comment]
Just to make sure I understand correctly. We would have to create rust bindings to similar to the [rust bindings that exist for `solc`](https://github.com/axic/solc-rust/blob/master/build.rs)?

**Comment by cburgdorf** [comment]
I also see that [yultsur](https://github.com/axic/yultsur) has compilation of YUL as planned feature so maybe it makes sense to contribute that upstream.

**Comment by g-r-a-n-t** [comment]
> Just to make sure I understand correctly. We would have to create rust bindings to similar to the rust bindings that exist for solc?

Yeah, that's what I'm hoping we can do. Although, there are alot of unknowns in my mind. I'm not even really sure what we would be binding to in libyul, so there will probably need to be some communication with the Solidity team.

> I also see that yultsur has compilation of YUL as planned feature so maybe it makes sense to contribute that upstream.

Yup, we'll definitely want to do this. I think this issue can be closed when the following are complete.

- [ ] add [libyul](https://github.com/ethereum/solidity/tree/develop/libyul) bindings to [Yulstur](https://github.com/g-r-a-n-t/yultsur)
- [ ] replace use of solc-rust in the [evm](https://github.com/ethereum/fe/tree/master/compiler/src/evm) compiler pass with Yultsur's libyul bindings

**Comment by sbillig** [comment]
Defunct
